### PR TITLE
fix(36381): blur composer on edit menu action

### PR DESCRIPTION
**Pull request by jeremy-croff**
### Fixed Issues

$ https://github.com/Expensify/App/issues/36381
PROPOSAL: https://github.com/Expensify/App/issues/36381#issuecomment-1940394893

### Tests

- [x] Verify that no errors appear in the JS console

For web and desktop:
1. Open a report

3. Focus a composer
4. Hover a previous message and press the three dot menu
5. Ensure the composer is unfocused
6. Check that the keyboard arrows work for the popped out menu from the menu options

1. From a non focused composer
2. Hover a previous message and press the three dot menu
3. Ensure the composer is unfocused 
4. Check that the keyboard arrows work for the popped out menu from the menu options

1. From a focused composer
2. Hover a previous message and press the quick reaction emojis
3. Ensure the composer is stays focused

For mobile:
1. Focus the composer
2. Click on a previous message to see the menu actions
3. Close it
4. Ensure the composer is still focused

### Offline tests

When offline, perform the same steps to ensure the focus is lost when going to the three dot menu. 

### QA Steps

- [x] Verify that no errors appear in the JS console
1. Open a report

3. Focus a composer
4. Hover a previous message and press the three dot menu
5. Ensure the composer is unfocused
6. Check that the keyboard arrows work for the popped out menu from the menu options

1. From a focused composer
2. Hover a previous message and press the quick reaction emojis
3. Ensure the composer is stays focused

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/157416545/4890c18b-f500-4e8e-9bde-8244c2c9a0ad

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/157416545/0bd2043c-e3ce-4d1b-9464-c55712b58aa5

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/157416545/03d499b0-16de-422b-9082-e7cffac2bcd9

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/157416545/f14dc96a-b3f9-48f0-aac6-6c42194da69e

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/157416545/279aa3ef-9acc-4117-987d-a737f90616e6

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/157416545/905bfb6e-c5ce-4b9d-92fa-c0b18d63df77

</details>

### CODE REVIEW

**Comment by jeremy-croff** [comment]
Note: During my testing against latest main I observed that the android composer and ios composer behave differently after opening up the menu actions of a row. IOS keeps it focused, android does not. I confirmed this is unrelated to my changes.

File: src/components/BaseMiniContextMenuItem.tsx [hunk 1]
```diff
@@ -32,13 +32,17 @@ type BaseMiniContextMenuItemProps = {
      * Whether the button should be in the active state
      */
     isDelayButtonStateComplete: boolean;
+    /**
+     * Can be used to control the click event, and for example whether or not to lose focus from the composer when pressing the item
+     */
+    shouldPreventDefault?: boolean;
```
[the remark this answers is not in the document]
Comment (jeremy-croff) [reply src/components/BaseMiniContextMenuItem.tsx:38]: I do see the point.
Preventdefault is a JavaScript api so when keeping it vague I intended it to just mean call the preventDefault api on all click events that click this action.

But if we want it to indicate the user experience, but would preventDefaultClickEvent be a bit better?

[the remark this answers is not in the document]
Comment (jeremy-croff) [reply src/components/BaseMiniContextMenuItem.tsx]: Updated!

**Comment by jeremy-croff** [comment]
> @jeremy-croff Step 13 looks redundant in "Tests" + Step 3 looks redundant in "QA Tests", doesn't it?

I feel like step 13 is an important new behavior that demonstrates that since the menu items are now focused when you press the 3 dot menu, it also activates the arrow keys of the keyboard, allowing you to switch between them. This previously didn't work.

Step 3, as part of QA, is the main flow that's addressed in the ticket; I'm unclear why it's redundant.

**Comment by jeremy-croff** [comment]
> > I feel like step 13 is an important new behavior that demonstrates that since the menu items are now focused when you press the 3 dot menu
> 
> Doesn't it cover in step 5 & 9? Because from step 12, we can test step 13 without tapping on three-dot menu
> 
> > Step 3, as part of QA, is the main flow that's addressed in the ticket; I'm unclear why it's redundant.
> 
> I mentioned it incorrectly. I mean step 9

Yep! You're totally right. Thanks.

**Review by hoangzinh — APPROVED** [review]
LGTM

**Review by bondydaa — APPROVED** [review]

**Pull request merged**